### ISSUE TITLE
Unintelligible behavior without Play Services update

### ISSUE THREAD

**Issue by bubenheimer** [googlesamples/android-play-location#82]
It seems that the new Play Services Location APIs in 11.0.1 show somewhat unintelligible behavior when a device does not have the ability to update Play Services, e.g. the Android Emulator without Play Store (I tested on Android Emulator API level 23, which does not have 11.x Play Services yet). The code below never reaches the OnCompleteListener; the "Update Play Services" dialog shows up, but then a stacktrace is printed from Play Services code, and nothing else happens. No feedback available to app code about the situation as far as I can tell.

    LocationServices
            .getSettingsClient(fragment.getActivity())
            .checkLocationSettings(settingsRequest)
            .addOnCompleteListener(task -> {});

    06-14 23:05:03.245 4606-4606/com.example.something E/DialogRedirect: Failed to start resolution intent
                                                                        android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=market://details?id=com.google.android.gms&pcampaignid=gcore_11010000-d-com.example.something-21 flg=0x80000 pkg=com.android.vending }
                                                                            at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1798)
                                                                            at android.app.Instrumentation.execStartActivity(Instrumentation.java:1512)
                                                                            at android.app.Activity.startActivityForResult(Activity.java:3917)
                                                                            at android.support.v4.app.BaseFragmentActivityJB.startActivityForResult(BaseFragmentActivityJB.java:50)
                                                                            at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:79)
                                                                            at android.support.v4.app.ActivityCompatJB.startActivityForResult(ActivityCompatJB.java:30)
                                                                            at android.support.v4.app.ActivityCompat.startActivityForResult(ActivityCompat.java:146)
                                                                            at android.support.v4.app.FragmentActivity.startActivityFromFragment(FragmentActivity.java:937)
                                                                            at android.support.v4.app.FragmentActivity$HostCallbacks.onStartActivityFromFragment(FragmentActivity.java:1047)
                                                                            at android.support.v4.app.Fragment.startActivityForResult(Fragment.java:963)
                                                                            at android.support.v4.app.Fragment.startActivityForResult(Fragment.java:952)
                                                                            at com.google.android.gms.common.internal.zzw.zzrt(Unknown Source)
                                                                            at com.google.android.gms.common.internal.zzt.onClick(Unknown Source)
                                                                            at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:163)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                            at android.os.Looper.loop(Looper.java:148)
                                                                            at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                            at java.lang.reflect.Method.invoke(Native Method)
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by bubenheimer** [comment]
I see that the Fragment containing this code receives a random call to onActivityResult() with a result code of 65538. Guessing that this is related to the problem, but don't know what to make of it.

**Comment by bubenheimer** [comment]
Tracking it down more I see that eventually a notification is displayed, which ultimately triggers the listener.

**Comment by samtstern** [comment]
@bubenheimer seems like you mostly figured it out.  The emulator behavior is worse than you'll see in the wild because they're some of the only devices that have Google Play services but don't have Play Store access to update it, which breaks the update flow.  I think if you try it on a real device you'll see something much more sane.

And the `65538` code you're seeing is a result of how `onActivityResult` works with Fragments.  The request code is bit-shifted and combined with the fragment's id so that the Activity knows which child Fragment to dispatch the result to.

**Comment by bubenheimer** [comment]
@samtstern thanks for the final word on this. It looks like the emulator behavior essentially mimics the behavior of a user who dismisses the update dialog, so it seems useful for consistently testing this case.

I believe though that I've identified an issue in 11.0.0 with the `GoogleApiClient` approach when using `enableAutoManage`. If an update (resolution) dialog pops up and there is a configuration change, the dialog does not come back. I still have to use this approach in other places with more complex connection-based operations, so I wonder how to get it in front of the right people to be fixed. I've created a small project to reproduce the issue at https://github.com/bubenheimer/playservicesupdaterotationbug
